src/lib/client.ts

```diff
@@ -1169,44 +1131,26 @@ export default class BACnetClient extends TypedEventEmitter<BACnetClientEvents>
 		)
 		ReadPropertyMultiple.encode(buffer, propertiesArray)
 		this.sendBvlc(receiver, buffer)
-		this._addCallback(settings.invokeId, (err, data) => {
-			if (err) {
-				return void next(err)
-			}
-
-			const result = ReadPropertyMultiple.decodeAcknowledge(
-				data.buffer,
-				data.offset,
-				data.length,
-			)
-			if (!result) {
-				return void next(new Error('INVALID_DECODING'))
-			}
-			next(null, result)
-		})
+		const data = await this._addDeferred(settings.invokeId)
+		const result = ReadPropertyMultiple.decodeAcknowledge(
+			data.buffer,
+			data.offset,
+			data.length,
+		)
+		if (!result) {
+			throw new Error('INVALID_DECODING')
+		}
+		return result
 	}
 
 	/**
 	 * The writePropertyMultiple command writes multiple properties in multiple objects to a device.
 	 */
-	writePropertyMultiple(
-		address: BACNetAddress,
-		values: WritePropertyMultipleObject[],
-		callback: ErrorCallback,
-	): void
-	writePropertyMultiple(
-		address: BACNetAddress,
-		values: WritePropertyMultipleObject[],
-		options: ServiceOptions,
-		callback: ErrorCallback,
-	): void
-	writePropertyMultiple(
+	async writePropertyMultiple(
```

Comment:
I see you removed the signature that allowed to call this without options, I would add that and make options optional